### DISCUSSION TITLE
How to validate name of a function?

### DISCUSSION THREAD

**Discussion by itsramiel** [open-circle/valibot#817] [Q&A] [answered]
Hi, I would like to make sure that a variable is a function but also check that the name of the function is of a specific value.

How can I do that?

One solution I guess is to pipe function with custom and check in custom, but the issue is the value passed to custom is of type`unknown` and not `function` like this:
```ts
v.pipe(
    v.function()
    v.custom(fn => ...)
)
```
Or
```
v.custom(
    (value) => typeof value === 'function' && value.name === 'name'
  )
  ```
  
  but I was wondering if there is a better way?
  Perhaps `function` could accept a name?

**Comment by fabian-hiller** [comment, marked as answer]
You should use `check` instead of `custom`. Try it out in our [playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6MMDB8AFPTjjsAV1ZMYwNjICUAGnnjefJgGs5ChTOWqtIgHw4VTMa2Qg+I4aMLYIEQnvM4GWBUVCU+G2F7AAMAFX44Fzc4ELgAIgASAG8QsL4xYFYwJRhnVz4AXzSU9hg+ZAATOAhsdJ8INOiAuF16LQZ6KzUNVjgKZChtOCyKxhZauCg+MNJ4UQwxVGRsPgAFCdRZbn5BHTGJ-rm2VAhSfNIIAHMZJZWYS6A).

```ts
import * as v from 'valibot';

const Schema = v.pipe(
  v.function(),
  v.check(
    (func) => func.name === 'foo',
    (issue) => `The name is "${issue.input.name}" instead of "foo"`,
  ),
);
```

**Reply by fabian-hiller** [reply]
`custom` is a schema, while `check` is a validation action. Schemas always expect unknown input. You may want to take a look at our mental model: https://valibot.dev/guides/mental-model/

**Reply by itsramiel** [reply]
Alright, thank you 🙏
